firefox-ios/Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -71,55 +70,15 @@ class ClipboardBarDisplayHandler: NSObject, URLChangeDelegate {
 
     @objc
     private func appWillEnterForegroundNotification() {
-        sessionStarted = true
-        checkIfShouldDisplayBar()
-    }
-
-    private func observeURLForFirstTab(firstTab: Tab) {
-        if firstTab.webView == nil {
-            // Nothing to do; bail out.
-            firstTabLoaded = true
-            return
-        }
-        self.firstTab = firstTab
-        firstTab.observeURLChanges(delegate: self)
-    }
-
-    func didRestoreSession() {
-        guard !sessionRestored else { return }
-        if let tabManager = self.tabManager,
-            let firstTab = tabManager.selectedTab {
-            observeURLForFirstTab(firstTab: firstTab)
-        } else {
-            firstTabLoaded = true
-        }
-
-        sessionRestored = true
-        checkIfShouldDisplayBar()
-    }
-
-    func tab(_ tab: Tab, urlDidChangeTo url: URL) {
-        // Ugly hack to ensure we wait until we're finished restoring the session on the first tab
-        // before checking if we should display the clipboard bar.
-        guard sessionRestored,
-            !url.absoluteString.hasPrefix("\(WebServer.sharedInstance.base)/about/sessionrestore?history=") else {
-            return
-        }
-
-        tab.removeURLChangeObserver(delegate: self)
-        firstTabLoaded = true
         checkIfShouldDisplayBar()
     }
 
     private func shouldDisplayBar(_ copiedURL: String) -> Bool {
-        if !sessionStarted ||
-            !sessionRestored ||
-            !firstTabLoaded ||
+        if !AppEventQueue.hasSignalled(.tabRestoration(windowUUID)) ||
```

Comment:
So we don't want to wait for the Event to happen, but rather do a one-time check whenever we launch the app?

Reply:
oh sorry I didn't push up the latest, wanted to do more testing first so I left it as a draft PR for now.